packages/api/src/controllers/asset.ts

```diff
@@ -1090,6 +1091,40 @@ app.delete("/:id", authorizer({}), async (req, res) => {
   res.end();
 });
 
+app.delete("/", authorizer({ anyAdmin: true }), async (req, res) => {
+  if (req.query.userId && req.query.userId !== req.user.id && !req.user.admin) {
+    throw new ForbiddenError(`users may only delete their own assets`);
+  }
+
+  if (!req.query.userId) {
+    throw new BadRequestError(`userId is required`);
+  }
+
+  const userId: string = req.query.userId.toString();
+
+  const user = await db.user.get(userId);
+
+  if (!user) {
+    throw new NotFoundError(`user not found`);
+  }
+
+  const [assets] = await db.asset.find([
```

Comment:
Notice this will only delete at most 1000 assets, not all of them. There is an enforced page limit on the DB code.